### DISCUSSION TITLE
cannot hide modal with javascript

### DISCUSSION THREAD

**Discussion by iqbalme** [coreui/coreui#486] [Q&A]
Hi, I have a modal like this:

```
<!-- Button trigger modal -->
<button type="button" class="btn btn-primary" data-coreui-toggle="modal" data-coreui-target="#exampleModal">
  Launch demo modal
</button>

<!-- Modal -->
<div class="modal fade" id="exampleModal" tabindex="-1" aria-labelledby="exampleModalLabel" aria-hidden="true">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title" id="exampleModalLabel">Modal title</h5>
        <button type="button" class="btn-close" data-coreui-dismiss="modal" aria-label="Close"></button>
      </div>
      <div class="modal-body">
        ...
      </div>
      <div class="modal-footer">
        <button type="button" class="btn btn-secondary" data-coreui-dismiss="modal">Close</button>
        <button type="button" class="btn btn-primary" onclick="closeModal()">Save changes</button>
      </div>
    </div>
  </div>
</div>
```

I want to hide it when I click save changes, so I add onclick event there and put this function:
```
<script>
function closeModal(){
const myModal = new coreui.Modal('#exampleModal');
myModal.hide();
}
</script>
```

But when I click the button, modal dissappeared but the background is still in dark as shown in this picture: https://prnt.sc/s6o5KD5LtGhh

Please can anyone help me about this? Thank you.

**Comment by shuvro111** [comment]
@iqbalme You have to use the `data-coreui-dismiss="modal"` attribute to the **Save Changes** button. You don't need to extra Javascript function to close the modal. However, if you want to execute  a block of Javascript code before closing the modal, you can also do that by using the `onclick="myfunction()"`

Here's a working example using [Core Ui](https://coreui.io/) version 4 👉  [Open In CodeSandbox](https://codesandbox.io/embed/bold-star-hbq5z3?fontsize=14&hidenavigation=1&theme=dark)

Hope that helps!
